src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -154,9 +154,8 @@ void primitive_inst::update_shape() {
         }
     }
 
-    // We assume that tensor ranks are static, thus shape_of doesn't need to update anything even if input shape is dynamic
-    if (_node.is_type<shape_of>() && !input_shape_changed)
-        return;
+    if (input_shape_changed)
+        set_shape_change();
```

Comment:
**input_shape_changed** can be changed by the following code. Is it okay to move this code forward?

Reply:
The next set of "input_shape_changed" is to check the memory dep, does not mean that it is really changed. 
We only need to "set_shape_change" when the input is really changed.